mir_eval/segment.py

```diff
@@ -131,7 +132,7 @@ def validate_structure(reference_intervals, reference_labels,
         # Check only when intervals are non-empty
         if intervals.size > 0:
             # Make sure intervals start at 0
-            if not np.allclose(intervals[0, 0], 0.0):
+            if not np.allclose(intervals.min(), 0.0):
```

Comment:
Nope :grin: the slice construction dominates runtime for the range of values we care about.

```python
N = 100
x = np.random.randn(N, 2)

%timeit x.min()
100000 loops, best of 3: 1.9 µs per loop

%timeit x[:, 0].min()
100000 loops, best of 3: 1.92 µs per loop
```